File: implementations/go/examples/internal/application/integration-event/payment_completed_integration_event.go
package integrationevent

import "time"

// PaymentCompletedV1 is the Integration Event (public contract) the Payment
// BC exposes to external BCs. It is a thin contract carrying only the
// minimal information Account needs for the actual debit (withdraw)
// (accountId+amount) — it does not expose Payment's internal model such as
// ownerId/cardId. PaymentID is also included as the correlation key the
// Account BC uses for its idempotency check (Level 2 Ledger: duplicate
// referenceId check).
type PaymentCompletedV1 struct {
	PaymentID   string    `json:"paymentId"`
	AccountID   string    `json:"accountId"`
	Amount      int64     `json:"amount"`
	CompletedAt time.Time `json:"completedAt"`
}

func (PaymentCompletedV1) EventName() string { return "payment.completed.v1" }

File: implementations/nestjs/examples/src/payment/application/integration-event/payment-completed-integration-event.ts
// The Integration Event (a public contract) Payment BC exposes to external BCs.
// A thin contract carrying only the minimal information Account needs for the actual debit
// (withdraw) — accountId+amount — never exposing Payment's internal model like ownerId/cardId.
// paymentId is also carried as the correlation key Account BC uses for its idempotency check
// (a Level 2 Ledger: a referenceId duplicate check).
export class PaymentCompletedIntegrationEventV1 {
  public readonly eventName = 'payment.completed.v1' as const

  constructor(
    public readonly paymentId: string,
    public readonly accountId: string,
    public readonly amount: number,
    public readonly completedAt: string
  ) {}
}
